src/language-js/needs-parens.js

```diff
@@ -349,12 +349,7 @@ function needsParens(path, options) {
       }
 
     case "YieldExpression":
-      if (
-        parent.type === "UnaryExpression" ||
-        parent.type === "AwaitExpression" ||
-        parent.type === "TSAsExpression" ||
-        parent.type === "TSNonNullExpression"
```

Comment:
`UnaryExpression`, `TSAsExpression` and `TSNonNullExpression` are treated on https://github.com/prettier/prettier/blob/next/src/language-js/needs-parens.js#L368-L370 so this change no affects for format result.